apps/dashboard/src/components/subscribers/locale-select.tsx

```diff
@@ -1,71 +1,111 @@
-  required?: boolean;
-  onValueChange: (val: string) => void;
+  onChange: (val: string) => void;
 }) {
+  const [open, setOpen] = useState(false);
+  const currentCountryCode = value?.split('_')?.[1] as Country;
+  const CurrentFlag = currentCountryCode ? flags[currentCountryCode] : RiEarthLine;
+
   return (
-    <Select
-      value={value}
-      onValueChange={onValueChange}
-      disabled={disabled || readOnly}
-      required={required}
-      defaultValue={defaultOption}
-    >
-      <SelectTrigger className="focus:ring-stroke-strong group p-1.5 shadow-sm focus:ring-1">
-        <SelectValue
-          placeholder={
-            <div className="flex w-full items-center gap-2">
-              <div>
-                <RiEarthLine className="size-4" />
-              </div>
-              <TruncatedText className="text-sm">Locale</TruncatedText>
-            </div>
-          }
-          asChild
+    <Popover open={open} onOpenChange={setOpen}>
+      <PopoverTrigger asChild>
+        <Button
+          type="button"
+          variant="secondary"
+          mode="outline"
+          className="flex h-8 w-full items-center justify-between gap-1 rounded-lg border-r-0 px-3 focus:z-10"
+          disabled={disabled}
         >
-          <div className="flex w-full items-center gap-2">
-            {value && (
-              <>
-                <div>
-                  <CountryFlags name={value.split('_')?.[1]} className="size-4" />
-                </div>
-                <TruncatedText className="text-sm">{value}</TruncatedText>
-              </>
-            )}
+          <div className="flex max-w-full flex-1 items-center gap-1 overflow-hidden">
+            <span className="inline-flex gap-1">
+              {CurrentFlag && <CurrentFlag title={currentCountryCode} className="size-4" />}
+              {value || <span className="text-neutral-400">locale</span>}
+            </span>
+
+            <RiArrowDownSLine
+              className={cn('ml-auto size-4 opacity-50', disabled || readOnly ? 'hidden' : 'opacity-100')}
+            />
           </div>
-        </SelectValue>
-      </SelectTrigger>
-      <SelectContent className="p-1">
-        {locales.map((item) => (
-          <SelectItem key={item.langIso} value={item.langIso} className="px-2">
-            <div className="flex w-full items-center gap-2">
-              <div>
-                <CountryFlags name={item.alpha2} className="size-4" />
-              </div>
-              <TruncatedText className="text-sm">
-                {item.langIso} - {item.langName}
-              </TruncatedText>
-            </div>
-          </SelectItem>
-        ))}
-      </SelectContent>
-    </Select>
+        </Button>
+      </PopoverTrigger>
+      <PopoverContent className="w-[300px] rounded-lg border-t-0 p-0">
+        <Command>
+          <CommandInput placeholder="Search locale..." />
+          <CommandList>
+            <CommandEmpty>No locale found.</CommandEmpty>
```

Comment:
this shouldn't be scrollable
![Screenshot 2025-02-07 at 18 44 19](https://github.com/user-attachments/assets/3f1013a3-4496-4282-9e99-d95ac17e4855)

Reply:
same timezone